snappy/DiskCache_DiskCache.py

```diff
@@ -532,6 +533,20 @@ def evict(self, key):
         except:
             logger.log(logLevel.ERROR, "Unable to delete file evicted from cache: {}"
                        .format(toEvict.path))
+        self.removeEmptyCacheDirs(os.path.dirname(toEvict.path))
+
+    def removeEmptyCacheDirs(self, directory):
+        while directory != config['cachePath']:
+            # Going for the "easier to ask for forgiveness than permission" model:
+            # Try to remove directory. Check afterwards to see if there was an
+            # error indicating that the directory was not empty
+            try:
+                os.rmdir(directory)
+            except OSError as ex:
+                if ex.errno == errno.ENOTEMPTY:
+                    return
+                raise ex
```

Comment:
Instead of doing this which will change the traceback stack, you should just do:

```
raise
```

That'll re-raise the exception being currently handled without affecting `exc_info`.